include/hw/ssi/spi_gpio.h

```diff
@@ -37,7 +37,12 @@ struct SpiGpioState {
     SSIBus *spi;
     AspeedGPIOState *aspeed_gpio;
 
-    int clk, mosi, cs;
+    int mode;
+    int clk_counter;
+    uint32_t output_byte;
+    uint32_t input_byte;
+
+    int clk, mosi, cs, miso;
```

Comment:
```suggestion
    bool clk, mosi, cs, miso;
```
as we keep on use !! to get a bool value. let's use bool. 
true means 1(high), and false means 0 (low)